oak_remote_attestation/src/dice.rs

```diff
@@ -191,24 +186,22 @@ impl TryFrom<DiceData> for DiceBuilder {
 impl TryFrom<Stage0DiceData> for DiceData {
     type Error = anyhow::Error;
     fn try_from(value: Stage0DiceData) -> anyhow::Result<Self> {
-        let (platform, remote_attestation_report) =
-            if value.root_layer_evidence.tee_platform == TeePlatform::AmdSevSnp as u64 {
-                (
-                    TeePlatform::AmdSevSnp as u64,
-                    value.root_layer_evidence.remote_attestation_report
-                        [..AMD_SEV_SNP_ATTESTATION_REPORT_SIZE]
-                        .to_vec(),
-                )
-            } else {
-                (TeePlatform::Unspecified as u64, Vec::new())
-            };
-        let platform = platform.try_into().map_err(anyhow::Error::msg)?;
-        let eca_public_key = oak_dice::utils::cbor_encoded_bytes_to_vec(
-            &value.root_layer_evidence.eca_public_key[..],
-        )
-        .map_err(anyhow::Error::msg)?;
+        let platform: TeePlatform = value
+            .root_layer_evidence
+            .get_tee_platform()
+            .map_err(anyhow::Error::msg)?
+            .into();
+        let remote_attestation_report = value
+            .root_layer_evidence
+            .get_remote_attestation_report()
+            .map_err(anyhow::Error::msg)?
+            .to_vec();
+        let eca_public_key = value
+            .root_layer_evidence
+            .get_eca_public_key()
+            .map_err(anyhow::Error::msg)?;
         let root_layer = Some(RootLayerEvidence {
-            platform,
+            platform: platform as i32,
```

Comment:
Why is the conversion to i32 needed now?

Reply:
because that's how enums work in prost :(

Reply:
Ah yes, they have typesafe getters and setters, but the type itself is quite raw. Probably not worth trying to use the setter here anyways, right?

https://github.com/tokio-rs/prost#enumerations

Reply:
Yeah I'm not sure it is. Because we'd still need to provide some value when creating the struct. (Unless we create from default & then call setters for each value. But that means the type system no longer checks that we set a value for each property. 🤷‍♀️